crates/polars-ops/src/series/ops/arg_min_max.rs

```diff
@@ -167,14 +167,14 @@ where
 }
 
 pub(crate) fn arg_max_bool(ca: &BooleanChunked) -> Option<usize> {
-    if ca.null_count() == ca.len() {
+    let null_count = ca.null_count();
+    if null_count == ca.len() {
         None
-    }
-    // don't check for any, that on itself is already an argmax search
-    else if ca.null_count() == 0 && ca.chunks().len() == 1 {
-        let arr = ca.downcast_iter().next().unwrap();
-        let mask = arr.values();
-        Some(first_set_bit(mask))
+    } else if null_count == 0 {
+        // if no null values we only have True/False, which can be downcast to
+        // operate on as bitmap 1/0, allowing for a fast-path; if `first_non_null`
+        // returns None this implies all values are zero (eg: False), so we return 0
+        first_non_null(ca.downcast_iter().map(|arr| Some(arr.values()))).or(Some(0))
```

Comment:
Not sure I like re-using this function for this, in general I should just write proper min/max kernels. But this works so no need to remove it.